### DISCUSSION TITLE
Additional paper that uses coordinate descent for conditional computing

### DISCUSSION THREAD

**Discussion by smdrnks** [lucidrains/CoLT5-attention#4] [General]
I came across this paper that uses the same formulation for sparse selection as CoLT5: https://arxiv.org/abs/2304.04947

They also show a trick on how to scale epsilon to improve convergence of coordinate descent, see appendix D.2. Could maybe be worth to have as an option for your implementation.

**Comment by lucidrains** [comment]
@smdrnks omg! thank you! 🙏 will check this out

**Comment by lucidrains** [comment]
are you running experiments on this? In my tests, this paper seems to work! still waiting for a catch..

**Reply by lucidrains** [reply]
@smdrnks it is [done](https://github.com/lucidrains/CoLT5-attention/commit/52a95c306b969022c750fbd8b42ac08a5d06b7f3#diff-cdc5aa568034877dbacf7b4916df4bc2e56f3f43b4d256f1be76fb4a5f450891R27), both for the naive as well as the triton implementation!
